src/AblyProvider.tsx

```diff
@@ -66,8 +66,10 @@ export const AblyProvider = ({
         throw new Error('Provide client or options, not both');
     }
 
-    const realtime: Realtime =
-        client || new Realtime(options as Ably.Types.ClientOptions);
+    const realtime = useMemo(
+        () => client || new Realtime(options as Ably.Types.ClientOptions),
+        [client, options]
```

Comment:
I think it won't resolve the issue, I believe we can expect that users will create provider something like this

```
<AblyProvider options={{ key: 'key', clientId: 'clientId' }}>
```

And we still will be recreating Ably client each render (and close only first one)

Reply:
Updated based on our discussion 😄